packages/ui-extensions-server-kit/src/ExtensionServerClient/ExtensionServerClient.ts

```diff
@@ -220,8 +221,14 @@ export class ExtensionServerClient implements ExtensionServer.Client {
         localization,
         name:
           parsedTranslation && extension.name.startsWith('t:')
-            ? this._getLocalizedName(parsedTranslation, extension.name)
+            ? this._getLocalizedValue(parsedTranslation, extension.name)
             : extension.name,
+        ...(extension.description && {
+          description:
+            parsedTranslation && extension.description?.startsWith('t:')
+              ? this._getLocalizedValue(parsedTranslation, extension.description)
+              : extension.description,
+        }),
```

Comment:
probably don't need to check if there is a description, this should be safe:

```suggestion
          description:
            parsedTranslation && extension.description?.startsWith('t:')
              ? this._getLocalizedValue(parsedTranslation, extension.description)
              : extension.description,
```

Reply:
Wouldn't that end up having also a `description: undefined` value in the payload if there is no description?

Reply:
It wouldn't make a difference since `description` is always optional. Consumers of the payload will always need to check if it's present so it doesn't matter if it's `undefined` or not set.